src/test/llmtemplate_test.cpp

```diff
@@ -0,0 +1,332 @@
+    nodeResources->modelsPath = directoryPath;
+    LLMNodeResources::loadTextProcessor(nodeResources, nodeResources->modelsPath);
+
+    std::string finalPrompt = "";
+    std::string payloadBody = R"(
+        {
+            "model": "gpt",
+            "stream": false,
+            "messages": [{"role": "user", "content": "hello"}]
+        }
+    )";
+    std::string expectedOutput = " Hi, HELLO ";
+    ASSERT_EQ(applyChatTemplate(nodeResources->textProcessor, nodeResources->modelsPath, payloadBody, finalPrompt), true);
+    ASSERT_EQ(finalPrompt, expectedOutput);
+}
+
+TEST_F(LLMChatTemplateTest, ChatTemplateJinjaException) {
+    std::string jinjaTemplate = R"( {{ "Hi, " + messages[3]['content'] | upper }} )";
+    CreateJinjaConfig(jinjaTemplate);
+    std::shared_ptr<LLMNodeResources> nodeResources = std::make_shared<LLMNodeResources>();
+    nodeResources->modelsPath = directoryPath;
+    LLMNodeResources::loadTextProcessor(nodeResources, nodeResources->modelsPath);
+
+    std::string finalPrompt = "";
+    std::string payloadBody = R"(
+        {
+            "model": "gpt",
+            "stream": false,
+            "messages": [{"role": "user", "content": "hello"}]
+        }
+    )";
+    ASSERT_EQ(applyChatTemplate(nodeResources->textProcessor, nodeResources->modelsPath, payloadBody, finalPrompt), false);
+}
+
+TEST_F(LLMChatTemplateTest, ChatTemplateTokenizerDefault) {
+    std::string tokenizerJson = R"({
+    "bos_token": "</s>",
+    "eos_token": "</s>"
+    })";
+    CreateTokenizerConfig(tokenizerJson);
+    std::shared_ptr<LLMNodeResources> nodeResources = std::make_shared<LLMNodeResources>();
+    nodeResources->modelsPath = directoryPath;
+    LLMNodeResources::loadTextProcessor(nodeResources, nodeResources->modelsPath);
+
+    std::string finalPrompt = "";
+    std::string payloadBody = R"(
+        {
+            "model": "gpt",
+            "stream": false,
+            "messages": [{"role": "user", "content": "hello"}]
+        }
+    )";
+    std::string expectedOutput = "hello";
+    ASSERT_EQ(applyChatTemplate(nodeResources->textProcessor, nodeResources->modelsPath, payloadBody, finalPrompt), true);
+    ASSERT_EQ(finalPrompt, expectedOutput);
+}
+
+TEST_F(LLMChatTemplateTest, ChatTemplateTokenizerException) {
+    std::string tokenizerJson = R"({
+    "bos_token": "</s>",
+    "eos_token": "</s>",
+    })";
+    CreateTokenizerConfig(tokenizerJson);
+    std::shared_ptr<LLMNodeResources> nodeResources = std::make_shared<LLMNodeResources>();
+    nodeResources->modelsPath = directoryPath;
+    LLMNodeResources::loadTextProcessor(nodeResources, nodeResources->modelsPath);
+
+    std::string finalPrompt = "";
+    std::string payloadBody = R"(
+        {
+            "model": "gpt",
+            "stream": false,
+            "messages": [{"role": "user", "content": "hello"}]
+        }
+    )";
+    std::string expectedOutput = "Error: Chat template not loaded correctly, so it cannot be applied";
```

Comment:
Why can't we simply return an error from loadTextProcessor when the template is incorrect and not even initialize the pipeline? @rasapala @mzegla

Reply:
This way we could ensure we never encounter such error during runtime

Reply:
Because we wanted to have chat template functionality and pipeline itself kind of decoupled. 
Chat template is used only for chat completion endpoint and the pipeline should be operational even if we fail loading it. Administrator gets appropriate log messages.